### PR TITLE
Add gosec as presubmit job

### PR DESCRIPTION
**Pull request by xpivarc**

### CODE REVIEW

**Review by dhiller — APPROVED** [review]
/lgtm
/approve

File: github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -658,6 +658,47 @@ presubmits:
...
+    skip_report: true
+    optional: true
+    decorate: true
+    decoration_config:
+      timeout: 1h
+      grace_period: 5m
+    labels:
+      preset-dind-enabled: "true"
+      preset-docker-mirror: "true"
+    spec:
```
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:678]: please add

`preset-bazel-cache: "true"`

here, then you can remove the env vars as theya re all covered with the preset

**Review by dhiller — CHANGES_REQUESTED** [review]

**Comment by xpivarc** [comment]
/hold 
Lets [this](https://github.com/kubevirt/kubevirt/pull/4192) merge first.

File: github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 2]
```diff
@@ -658,6 +658,47 @@ presubmits:
           resources:
             requests:
               memory: "4Gi"
+
+  - name: pull-kubevirt-gosec
+    cluster: ibm-prow-jobs
+    skip_branches:
+      - release-\d+\.\d+
+    annotations:
+      fork-per-release: "true"
```
Comment (danielBelenky) [inline github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:667]: can you add the `"rehearsal.allowed": "true"` annotation? we then will be able to rehearse this job to see how it's working.

**Review by danielBelenky — CHANGES_REQUESTED** [review]

File: github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 2 above)
Comment (xpivarc) [reply github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:667]: done

File: github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 3]
```diff
@@ -658,6 +658,40 @@ presubmits:
...
+    optional: true
+    decorate: true
+    decoration_config:
+      timeout: 1h
+      grace_period: 5m
+    labels:
+      preset-dind-enabled: "true"
+      preset-docker-mirror: "true"
+      preset-bazel-cache: "true"
+      rehearsal.allowed: "true"
```
Comment (danielBelenky) [inline github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:679]: this should be an annotation, not a label :smile:

**Review by danielBelenky — CHANGES_REQUESTED** [review]

File: github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 3 above)
Comment (xpivarc) [reply github/ci/prow/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: My bad.

**Comment by danielBelenky** [comment]
hm.. looks like a bug in the rehearsal plugin :/ it runs the wrong jobs. I'll look into it

**Comment by dhiller** [comment]
/override rehearsal-pull-kubernetes-nmstate-e2e-handler-ocp
/override rehearsal-pull-kubernetes-nmstate-e2e-handler-ok
/override rehearsal-pull-kubevirt-generate

@danielBelenky seems like the bug in the rehearsal plugin still persists.

**Review by dhiller — COMMENTED** [review]
/lgtm
/approve
/hold cancel

**Review by dhiller — APPROVED** [review]

**Pull request merged**